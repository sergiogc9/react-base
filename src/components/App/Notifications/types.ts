import { Notification, QueueNotification } from '@src/types/notification';

export type StateProps = {
	notifications: QueueNotification[]
};

export type DispatchProps = {
	onAddNotification: (notification: Notification) => void,
	onUnqueueNotification: () => void
};

export type OwnProps = {};

export type ComponentProps = StateProps & DispatchProps & OwnProps;
