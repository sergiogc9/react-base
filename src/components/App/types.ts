import { TenantObject } from "@src/class/Tenant";
import { UserObject } from "@src/class/User";
import { FacebookAuthStatus } from "@src/types/facebook";
import { TProps } from '@src/lib/i18n';

export type StateProps = {
	authenticated: boolean
	tenant: TenantObject | null
	user: UserObject | null
};

export type DispatchProps = {
	onAuth: () => void
};

export type OwnProps = {};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
