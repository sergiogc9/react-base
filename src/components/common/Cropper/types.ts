type Props = {
	/**
	 * The image in Base64. If null, no image has been set before.
	 */
	readonly image: string;
	/**
	 * A callback method called with the cropped image data
	 */
	readonly onImageCropped: (image: string) => void;
};

export type CropperProps = Props;
