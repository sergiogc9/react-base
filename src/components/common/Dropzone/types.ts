import { TProps } from '@src/lib/i18n';

export type AddFileError = {
	error: 'file_too_large' | 'incorrect_file_type'
	file: File
};

export type StateProps = {};

export type DispatchProps = {};

export type OwnProps = {
	files: File[]
	multiple: boolean
	accept?: string
	maxSize?: number
	showError?: boolean
	onFilesAdded(files: File[]): void
	onAddFilesError(errors: AddFileError[]): void
	onRemoveFile(file: File): void
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
