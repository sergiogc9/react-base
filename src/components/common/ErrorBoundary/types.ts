export type ErrorBoundaryProps = React.PropsWithChildren;
