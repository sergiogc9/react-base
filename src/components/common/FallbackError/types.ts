import { FallbackProps } from 'react-error-boundary';

export type FallbackErrorProps = FallbackProps;
