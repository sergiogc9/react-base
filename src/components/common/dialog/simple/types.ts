import { TProps } from '@src/lib/i18n';

export type StateProps = {};

export type DispatchProps = {};

export type OwnProps = {
	id: string
	text: string
	content: JSX.Element
	onAccept?: () => void
	onAcceptText?: string
	onCancel?: () => void
	onCancelText?: string
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
