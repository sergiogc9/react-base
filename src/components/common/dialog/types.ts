import { TProps } from '@src/lib/i18n';

export type StateProps = {
};

export type DispatchProps = {
};

export type OwnProps = {
	id: string
	icon?: JSX.Element
	title: string
	content: JSX.Element
	onAccept?: () => void
	onAcceptText?: string
	onCancel?: () => void
	onCancelText?: string
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
