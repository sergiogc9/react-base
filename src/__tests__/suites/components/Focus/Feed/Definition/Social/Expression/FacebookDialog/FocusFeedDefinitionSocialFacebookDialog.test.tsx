import React from "react";
import { shallow } from "enzyme";

import FocusFeedDefinitionSocialFacebookDialog from "@src/components/Focus/Feed/Definition/Social/FacebookDialog/FocusFeedDefinitionSocialFacebookDialog";
import { InstagramAccount, SearchInstagramAccount, SearchFacebookPage } from "@src/types/facebook";

const instagramAccount: InstagramAccount = {
	id: "id1",
	facebook_page_id: "facebook-page-id",
	facebook_page_name: "facebook-page-name",
	name: "Girona FC",
	profile_picture_url: "fakeurl.com",
	screen_name: "gironafc"
};

const searchInstagramAccount: SearchInstagramAccount = {
	id: "searchIg1",
	ig_id: "23i92234",
	name: "IG account",
	followers_count: 100,
	follows_count: 50,
	media_count: 10,
	username: "fakeuser",
	profile_picture_url: "fakeurl.com",
	biography: "bio",
	page: {
		id: "page1",
		name: "Facebook page",
		access_token: "token!",
		picture: { data: { url: "fakeurlpage.com" } }
	}
};

const facebookPage: SearchFacebookPage = {
	id: searchInstagramAccount.page.id,
	name: searchInstagramAccount.page.name,
	picture: searchInstagramAccount.page.picture.data.url,
	instagramAccounts: [searchInstagramAccount, { ...searchInstagramAccount, id: instagramAccount.id }]
};

describe("Definition social facebook dialog", () => {
	let wrapper: any;
	let onCloseMock = jest.fn();
	let onAuthorizeSelectedAccountsMock = jest.fn();
	let onResetDataMock = jest.fn();
	let onSetFacebookUserIdMock = jest.fn();
	let onToggleSelectedAccountMock = jest.fn();

	const getComponent = () => (
		<FocusFeedDefinitionSocialFacebookDialog
			facebookUserId={"1234"}
			instagramAccounts={[instagramAccount]}
			facebookPages={[facebookPage]}
			selectedAccounts={[]}
			onAuthorizeSelectedAccounts={onAuthorizeSelectedAccountsMock}
			onResetData={onResetDataMock}
			onSetFacebookUserId={onSetFacebookUserIdMock}
			onToggleSelectedAccount={onToggleSelectedAccountMock}
			onClose={onCloseMock}
		/>
	);

	beforeEach(() => {
		onCloseMock = jest.fn();
		onAuthorizeSelectedAccountsMock = jest.fn();
		onResetDataMock = jest.fn();
		onSetFacebookUserIdMock = jest.fn();
		onToggleSelectedAccountMock = jest.fn();
		wrapper = shallow(getComponent());
	});

	it("Component default snaphsot", () => {
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Component snapshot without facebook user id", () => {
		wrapper.setProps({ facebookUserId: null });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Component snapshot when searching accounts", () => {
		wrapper.setProps({ instagramAccounts: null, facebookPages: null });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Component snapshot with no facebook pages", () => {
		wrapper.setProps({ facebookPages: [] });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Component close btn handler", () => {
		expect(onCloseMock).toHaveBeenCalledTimes(0);
		expect(onResetDataMock).toHaveBeenCalledTimes(0);
		wrapper.find('FontIcon').at(0).simulate('click');
		expect(onCloseMock).toHaveBeenCalledTimes(1);
		expect(onResetDataMock).toHaveBeenCalledTimes(1);
	});

	it("Dialog onHide handler", () => {
		expect(onCloseMock).toHaveBeenCalledTimes(0);
		expect(onResetDataMock).toHaveBeenCalledTimes(0);
		wrapper.find('DialogContainer').at(0).props().onHide();
		expect(onCloseMock).toHaveBeenCalledTimes(1);
		expect(onResetDataMock).toHaveBeenCalledTimes(1);
	});

	it("Dialog on cancel button handler", () => {
		expect(onCloseMock).toHaveBeenCalledTimes(0);
		expect(onResetDataMock).toHaveBeenCalledTimes(0);
		wrapper.find('#focusFeedDefinitionSocialFacebookDialogCancelBtn').at(0).simulate('click');
		expect(onCloseMock).toHaveBeenCalledTimes(1);
		expect(onResetDataMock).toHaveBeenCalledTimes(1);
	});

	it("Dialog on accept button handler with empty selected accounts", () => {
		expect(onCloseMock).toHaveBeenCalledTimes(0);
		expect(onResetDataMock).toHaveBeenCalledTimes(0);
		expect(onAuthorizeSelectedAccountsMock).toHaveBeenCalledTimes(0);
		wrapper.find('#focusFeedDefinitionSocialFacebookDialogAcceptBtn').at(0).simulate('click');
		expect(onCloseMock).toHaveBeenCalledTimes(1);
		expect(onResetDataMock).toHaveBeenCalledTimes(1);
		expect(onAuthorizeSelectedAccountsMock).toHaveBeenCalledTimes(0);
	});

	it("Dialog on accept button handler with selected accounts", () => {
		wrapper.setProps({ selectedAccounts: ["igId"] });
		expect(onCloseMock).toHaveBeenCalledTimes(0);
		expect(onResetDataMock).toHaveBeenCalledTimes(0);
		expect(onAuthorizeSelectedAccountsMock).toHaveBeenCalledTimes(0);
		wrapper.find('#focusFeedDefinitionSocialFacebookDialogAcceptBtn').at(0).simulate('click');
		expect(onCloseMock).toHaveBeenCalledTimes(1);
		expect(onResetDataMock).toHaveBeenCalledTimes(1);
		expect(onAuthorizeSelectedAccountsMock).toHaveBeenCalledTimes(1);
	});

	it("Dialog on facebook auth logged", () => {
		wrapper.setProps({ facebookUserId: null });
		expect(onSetFacebookUserIdMock).toHaveBeenCalledTimes(0);
		wrapper.find('Connect(FacebookAuth)').at(0).props().onLogin({ userID: "1234" });
		expect(onSetFacebookUserIdMock).toHaveBeenCalledTimes(1);
		expect(onSetFacebookUserIdMock).toHaveBeenCalledWith("1234");
	});

	it("Dialog on toggle selected account", () => {
		expect(onToggleSelectedAccountMock).toHaveBeenCalledTimes(0);
		wrapper.find('#focusFeedDefinitionSocialFacebookDialogInstagramCheckbox' + searchInstagramAccount.id).at(0).props().onChange();
		expect(onToggleSelectedAccountMock).toHaveBeenCalledTimes(1);
	});
});
