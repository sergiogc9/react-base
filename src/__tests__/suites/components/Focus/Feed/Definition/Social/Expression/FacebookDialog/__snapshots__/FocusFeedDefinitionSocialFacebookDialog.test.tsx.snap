// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition social facebook dialog Component default snaphsot 1`] = `
"<DialogContainer id=\\"focusFeedDefinitionSocialFacebookDialog\\" visible={true} onHide={[Function]} aria-labelledby=\\"focusFeedDefinitionSocialFacebookDialog\\" focusOnMount={false} autopadContent={true} autosizeContent={true} component=\\"span\\" closeOnEsc={true} contentComponent=\\"section\\" transitionEnterTimeout={300} transitionLeaveTimeout={300} defaultVisibleTransitionable={false} activeElementFocus={true}>
  <div className=\\"dialog-content\\">
    <div className=\\"dialog-close-button\\">
      <FontIcon onClick={[Function]} iconClassName=\\"material-icons\\">
        clear
      </FontIcon>
    </div>
    <div className=\\"dialog-header\\">
      <div className=\\"dialog-header-icon\\">
        <FontIcon iconClassName=\\"icon-instagram\\" />
      </div>
      <div className=\\"dialog-header-title\\">
        facebook_login.instagram.title
      </div>
    </div>
    <div className=\\"dialog-content-body\\">
      <div className=\\"dialog-content-body-text\\">
        facebook_login.instagram.description
      </div>
      <div className=\\"dialog-content-body-facebook-header\\">
        <FontIcon iconClassName=\\"icon-facebook\\" />
        <span>
          facebook_login.instagram.facebook_pages_label
        </span>
      </div>
      <div className=\\"dialog-content-body-facebook-page\\">
        <div className=\\"dialog-content-body-facebook-page-info\\">
          <Memo() url=\\"fakeurlpage.com\\">
            <FontIcon iconClassName=\\"icon-facebook\\" />
          </Memo()>
          <span>
            Facebook page
          </span>
        </div>
        <div className=\\"page-info-instagram-text\\">
          facebook_login.instagram.page_related_profiles_label
        </div>
        <div className=\\"dialog-content-body-facebook-page-instagram-account\\">
          <Memo() url=\\"fakeurl.com\\">
            <FontIcon iconClassName=\\"icon-instagram\\" />
          </Memo()>
          <div className=\\"dialog-content-body-facebook-page-instagram-account-info\\">
            <div className=\\"title\\">
              IG account
            </div>
            <div>
              @
              fakeuser
            </div>
          </div>
          <SelectionControl id=\\"focusFeedDefinitionSocialFacebookDialogInstagramCheckboxsearchIg1\\" type=\\"checkbox\\" label=\\"facebook_login.instagram.connect\\" name=\\"focusFeedHeaderToggleFeedEnabled\\" checked={false} disabled={false} onChange={[Function: onChange]} checkedCheckboxIcon={{...}} uncheckedCheckboxIcon={{...}} checkedRadioIcon={{...}} uncheckedRadioIcon={{...}} />
        </div>
        <div className=\\"dialog-content-body-facebook-page-instagram-account\\">
          <Memo() url=\\"fakeurl.com\\">
            <FontIcon iconClassName=\\"icon-instagram\\" />
          </Memo()>
          <div className=\\"dialog-content-body-facebook-page-instagram-account-info\\">
            <div className=\\"title\\">
              IG account
            </div>
            <div>
              @
              fakeuser
            </div>
          </div>
          <SelectionControl id=\\"focusFeedDefinitionSocialFacebookDialogInstagramCheckboxid1\\" type=\\"checkbox\\" label=\\"facebook_login.instagram.connected\\" name=\\"focusFeedHeaderToggleFeedEnabled\\" checked={true} disabled={true} onChange={[Function: onChange]} checkedCheckboxIcon={{...}} uncheckedCheckboxIcon={{...}} checkedRadioIcon={{...}} uncheckedRadioIcon={{...}} />
        </div>
      </div>
    </div>
    <div className=\\"dialog-footer\\">
      <withInk(withTooltip(Button)) flat={true} id=\\"focusFeedDefinitionSocialFacebookDialogCancelBtn\\" onClick={[Function]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
        form.cancel
      </withInk(withTooltip(Button))>
      <withInk(withTooltip(Button)) flat={true} id=\\"focusFeedDefinitionSocialFacebookDialogAcceptBtn\\" onClick={[Function]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
        form.accept
      </withInk(withTooltip(Button))>
    </div>
  </div>
</DialogContainer>"
`;

exports[`Definition social facebook dialog Component snapshot when searching accounts 1`] = `
"<DialogContainer id=\\"focusFeedDefinitionSocialFacebookDialog\\" visible={true} onHide={[Function]} aria-labelledby=\\"focusFeedDefinitionSocialFacebookDialog\\" focusOnMount={false} autopadContent={true} autosizeContent={true} component=\\"span\\" closeOnEsc={true} contentComponent=\\"section\\" transitionEnterTimeout={300} transitionLeaveTimeout={300} defaultVisibleTransitionable={false} activeElementFocus={true}>
  <div className=\\"dialog-content\\">
    <div className=\\"dialog-close-button\\">
      <FontIcon onClick={[Function]} iconClassName=\\"material-icons\\">
        clear
      </FontIcon>
    </div>
    <div className=\\"dialog-header\\">
      <div className=\\"dialog-header-icon\\">
        <FontIcon iconClassName=\\"icon-instagram\\" />
      </div>
      <div className=\\"dialog-header-title\\">
        facebook_login.instagram.title
      </div>
    </div>
    <div className=\\"dialog-content-body\\">
      <div className=\\"dialog-content-body-text\\">
        facebook_login.instagram.description
      </div>
      <CircularProgress id=\\"focusFeedDefinitionSocialFacebookDialogLoading\\" style={{...}} scale={1} determinateDashoffset={187} centered={true} />
    </div>
  </div>
</DialogContainer>"
`;

exports[`Definition social facebook dialog Component snapshot with no facebook pages 1`] = `
"<DialogContainer id=\\"focusFeedDefinitionSocialFacebookDialog\\" visible={true} onHide={[Function]} aria-labelledby=\\"focusFeedDefinitionSocialFacebookDialog\\" focusOnMount={false} autopadContent={true} autosizeContent={true} component=\\"span\\" closeOnEsc={true} contentComponent=\\"section\\" transitionEnterTimeout={300} transitionLeaveTimeout={300} defaultVisibleTransitionable={false} activeElementFocus={true}>
  <div className=\\"dialog-content\\">
    <div className=\\"dialog-close-button\\">
      <FontIcon onClick={[Function]} iconClassName=\\"material-icons\\">
        clear
      </FontIcon>
    </div>
    <div className=\\"dialog-header\\">
      <div className=\\"dialog-header-icon\\">
        <FontIcon iconClassName=\\"icon-instagram\\" />
      </div>
      <div className=\\"dialog-header-title\\">
        facebook_login.instagram.title
      </div>
    </div>
    <div className=\\"dialog-content-body\\">
      <div className=\\"dialog-content-body-text\\">
        facebook_login.instagram.description
      </div>
      <div className=\\"dialog-content-body-facebook-empty\\">
        facebook_login.instagram.no_results
      </div>
    </div>
    <div className=\\"dialog-footer\\">
      <withInk(withTooltip(Button)) flat={true} id=\\"focusFeedDefinitionSocialFacebookDialogCancelBtn\\" onClick={[Function]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
        form.cancel
      </withInk(withTooltip(Button))>
      <withInk(withTooltip(Button)) flat={true} id=\\"focusFeedDefinitionSocialFacebookDialogAcceptBtn\\" onClick={[Function]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300}>
        form.accept
      </withInk(withTooltip(Button))>
    </div>
  </div>
</DialogContainer>"
`;

exports[`Definition social facebook dialog Component snapshot without facebook user id 1`] = `
"<DialogContainer id=\\"focusFeedDefinitionSocialFacebookDialog\\" visible={true} onHide={[Function]} aria-labelledby=\\"focusFeedDefinitionSocialFacebookDialog\\" focusOnMount={false} autopadContent={true} autosizeContent={true} component=\\"span\\" closeOnEsc={true} contentComponent=\\"section\\" transitionEnterTimeout={300} transitionLeaveTimeout={300} defaultVisibleTransitionable={false} activeElementFocus={true}>
  <div className=\\"dialog-content\\">
    <div className=\\"dialog-close-button\\">
      <FontIcon onClick={[Function]} iconClassName=\\"material-icons\\">
        clear
      </FontIcon>
    </div>
    <div className=\\"dialog-header\\">
      <div className=\\"dialog-header-icon\\">
        <FontIcon iconClassName=\\"icon-instagram\\" />
      </div>
      <div className=\\"dialog-header-title\\">
        facebook_login.instagram.title
      </div>
    </div>
    <div className=\\"dialog-content-body\\">
      <div className=\\"dialog-content-body-description\\">
        <p>
          facebook_login.instagram.description_1
          <br />
          facebook_login.instagram.description_2
        </p>
        <p>
          facebook_login.instagram.description_3
        </p>
        <p>
          facebook_login.instagram.description_4
        </p>
      </div>
      <div className=\\"dialog-content-body-facebook-button\\">
        <Connect(FacebookAuth) permissions={{...}} onLogin={[Function]} />
      </div>
    </div>
  </div>
</DialogContainer>"
`;
